lightning-macros/src/lib.rs

```diff
@@ -74,3 +77,175 @@ pub fn maybe_await(expr: TokenStream) -> TokenStream {
+			let co2 = arm_stream.next().unwrap();
+			expect_punct(&co2, ':');
+			let variant_ident = arm_stream.next().unwrap();
+			let fields = arm_stream.next().unwrap();
+			let eq = arm_stream.next().unwrap();
+			expect_punct(&eq, '=');
+			let gt = arm_stream.next().unwrap();
+			expect_punct(&gt, '>');
+			let init = arm_stream.next().unwrap();
+
+			let next_tok = arm_stream.peek();
+			if let Some(TokenTree::Punct(_)) = next_tok {
+				expect_punct(next_tok.unwrap(), ',');
+				arm_stream.next();
+			}
+
+			let mut computed_fields = proc_macro::TokenStream::new();
+			if let TokenTree::Group(group) = fields {
+				if group.delimiter() == Delimiter::Brace {
+					let mut fields_stream = group.stream().into_iter().peekable();
+
+					let mut new_fields = proc_macro::TokenStream::new();
+					loop {
+						let next_tok = fields_stream.peek();
+						if let Some(TokenTree::Punct(_)) = next_tok {
+							let dot1 = fields_stream.next().unwrap();
+							expect_punct(&dot1, '.');
+							let dot2 = fields_stream.next().expect("Missing second trailing .");
+							expect_punct(&dot2, '.');
+							let trailing_dots = [dot1, dot2];
+							new_fields.extend(trailing_dots.into_iter().map(token_to_stream));
+							assert!(fields_stream.peek().is_none());
+							break;
+						}
+
+						let ref_ident = fields_stream.next().unwrap();
+						expect_ident(&ref_ident, Some("ref"));
+						let field_name_ident = fields_stream.next().unwrap();
+						let co = fields_stream.next().unwrap();
+						expect_punct(&co, ':');
+						let ty_info = fields_stream.next().unwrap();
+						let com = fields_stream.next().unwrap();
+						expect_punct(&com, ',');
+
+						if let TokenTree::Group(group) = ty_info {
+							let first_group_tok = group.stream().into_iter().next().unwrap();
+							if let TokenTree::Ident(ident) = first_group_tok {
+								if ident.to_string() == "legacy" {
+									continue;
+								}
+							}
+						}
+
+						let field = [ref_ident, field_name_ident, com];
+						new_fields.extend(field.into_iter().map(token_to_stream));
+					}
+					let fields_group = Group::new(Delimiter::Brace, new_fields);
+					computed_fields.extend(token_to_stream(TokenTree::Group(fields_group)));
+				} else {
+					computed_fields.extend(token_to_stream(TokenTree::Group(group)));
+				}
+			}
+
+			let arm_pfx = [enum_ident, co1, co2, variant_ident];
+			new_arms.extend(arm_pfx.into_iter().map(token_to_stream));
+			new_arms.extend(computed_fields);
+			let arm_sfx = [eq, gt, init];
+			new_arms.extend(arm_sfx.into_iter().map(token_to_stream));
+		}
+
+		let new_arm_group = Group::new(Delimiter::Brace, new_arms);
+		res.extend(token_to_stream(TokenTree::Group(new_arm_group)));
+	} else {
+		panic!("Expected `match self {{..}}` and nothing else");
+	}
+
+	assert!(stream.next().is_none(), "Expected `match self {{..}}` and nothing else");
+
+	res
+}
+
+/// Scans an enum definition for fields initialized to `LDK_DROP_LEGACY_FIELD_DEFINITION` and drops
```

Comment:
Reading through the code, I wasn’t quite able to figure out how `LDK_DROP_LEGACY_FIELD_DEFINITION` will be used in the end. I’d love to get some insights on that! Thanks!